Backend/Services/UserService.cs

```diff
@@ -0,0 +1,182 @@
+                        _User.Password = Encoding.UTF8.GetString(ByteArray, 0, ByteArray.Length);
+                    }
+                    await UnitOfWork.UserRepository.Create(_User, Cancel);
+                    if (_User.UserRoles == null)
+                        _User.UserRoles = new List<UserRole>();
+                    if (_User.UserAchievements == null)
+                        _User.UserAchievements = new List<UserAchievement>();
+                    foreach (var Role in User.Roles) 
+                    {
+                        _User.UserRoles.Add(await GenerateUserRole(_User, Role, Cancel));
+                    }
+                   
+
+                    await UnitOfWork.SaveChangesAsync(Cancel);
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                
+            }
+            catch (Exception exc)
+            {
+                throw exc;
+            }
+        }
+
+        public async Task<bool> DeleteUser(Guid Id, CancellationToken Cancel)
+        {
+            try
+            {
+                if ((await UnitOfWork.UserRepository
+                .DbSet.FirstOrDefaultAsync(item => item.Id == Id)) != null)
+                {
+                    await UnitOfWork.UserRepository.Delete(Id, Cancel);
+                    await UnitOfWork.SaveChangesAsync(Cancel);
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception exc)
+            {
+
+                throw exc;
+            }
+        }
+
+        public async Task<UserReadDTO> GetUserById(Guid Id, CancellationToken Cancel)
+        {
+            return Mapper.Map<UserReadDTO>(await ((UnitOfWork)UnitOfWork)._Context.Users
+                .Include(User => User.UserRoles)
+                .ThenInclude(UserRole => UserRole.Role)
+                .Include(User => User.UserAchievements)
+                .ThenInclude(UserAchievement => UserAchievement.Achievement)
+                .FirstOrDefaultAsync(User => User.Id == Id , Cancel));
+        }
+
+        public async Task<IEnumerable<UserReadDTO>> GetAllUsers(CancellationToken Cancel)
+        {
+            return Mapper.Map< IEnumerable<UserReadDTO>>(await ((UnitOfWork)UnitOfWork)._Context.Users
+                .Include(User => User.UserRoles)
+                .ThenInclude(UserRole => UserRole.Role)
+                .Include(User => User.UserAchievements)
+                .ThenInclude(UserAchievement => UserAchievement.Achievement)
+                .ToListAsync(Cancel));
+        }
+
+        public async Task<IEnumerable<UserReadShortDTO>> GetAllUsersInfo(CancellationToken Cancel)
+        {
+            return Mapper.Map<IEnumerable<UserReadShortDTO>>(await((UnitOfWork)UnitOfWork)._Context.Users
+               .Include(User => User.UserRoles)
```

Comment:
Why here you are using the Context from UnitOfWork but not UserRepository. Please use UserRepository for that needs

Reply:
Fixed